pkg/actions/lua/lakefs/catalogexport/symlink_exporter.lua

```diff
@@ -0,0 +1,93 @@
+local extractor = require("lakefs/catalogexport/table_extractor")
+local hive = require("lakefs/catalogexport/hive")
+local utils = require("lakefs/catalogexport/internal")
+local url = require("net/url")
+local pathlib = require("path")
+local strings = require("strings")
+
+-- make it a function so that @nopcoder can ask why we bloat the code? 
+local function parse_storage_uri(uri)
+    local parsed_uri = url.parse(uri)
+    local storage = parsed_uri.scheme
+    local bucket = parsed_uri.host
+    local key = parsed_uri.path
+    if key ~= nil then
+        key = key:sub(2) -- remove the leading slash
+    end
+    return storage, bucket, key
+end
+
+--[[
+    ### Default Symlink File(s) structure:
+    
+    ${storageNamespace}
+    _lakefs/
+        exported/
+            ${ref}/
+                ${commitId}/
+                    ${tableName}/
+                        p1=v1/symlink.txt
+                        p1=v2/symlink.txt
+                        p1=v3/symlink.txt
+                        ...
+]]
+local function gen_storage_uri_prefix(storage_ns, commit_id, action_info)
```

Comment:
1. Why this function exposed? we don't call it here and prefer not to give it to our users it looks internal as it writes under `_lakefs`. looks like something our code need to format just before writting the data
2. If we plan the user too call it and pass it to one of writers, I suggest accept the action_info in the writer and extract the data+format the storage ns and everything you need to know the default layout under the repository to write to.
3. If it just string manipulation and not generate anything I suggest format or fmt.

Reply:
1. 👍 
2. 👍  
3. 👍